portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +27,34 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+private ArrayList<String> messages = new ArrayList<>();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    response.getWriter().println("<h1>Hello Shengjing Zhang!</h1>");
+
+    Gson gson = new Gson();
+    String json = gson.toJson(messages);
+
+    // Send the JSON as the response
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String comments = request.getParameter("comment-input");
+    messages.add(comments);
+
+    response.setContentType("text/html");
+    response.getWriter().println(comments);
+    response.getWriter().println("<br/><br/>");
+    response.getWriter().println("Thanks for your comment!");
```

Comment:
Consider using String formatter or string concatenation here, because response.getWriter should potentially add some performance lag.